browser/request_otr/request_otr_browsertest.cc

```diff
@@ -0,0 +1,365 @@
+
+  infobar_manager->RemoveObserver(&observer);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest, ShowInterstitialAndProceedOTR) {
+  ASSERT_TRUE(InstallMockExtension());
+
+  // If request-otr pref is 'ask', should show interstitial because
+  // this URL is included in config file. If user clicks 'Proceed
+  // Off-The-Record' then we should navigate to the originally requested page
+  // and show an infobar indicating to the user that they are navigating in
+  // off-the-record mode.
+  auto* model = browser()->tab_strip_model();
+  auto* contents = model->GetActiveWebContents();
+  auto* infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(contents);
+  TestObserver observer;
+  // Set up expectation that an infobar will appear later.
+  EXPECT_CALL(observer, OnInfoBarAdded(_)).Times(1);
+  infobar_manager->AddObserver(&observer);
+
+  SetRequestOTRPref(RequestOTRService::RequestOTRActionOption::kAsk);
+  GURL url = embedded_test_server()->GetURL("sensitive.a.com", "/simple.html");
+  NavigateTo(url);
+  ASSERT_TRUE(IsShowingInterstitial());
+
+  // Simulate click on "Proceed Off-The-Record" button. This should navigate to
+  // the originally requested page in off-the-record mode.
+  ClickAndWaitForNavigation("primary-button");
+  ASSERT_FALSE(IsShowingInterstitial());
+  // Request-OTR infobar should now have been shown, and our observer should
+  // have been called once.
+
+  infobar_manager->RemoveObserver(&observer);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest,
+                       ShowInterstitialAndProceedNormally) {
+  ASSERT_TRUE(InstallMockExtension());
+
+  // If request-otr pref is 'ask', we should show an interstitial because this
+  // URL is included in config file. If user clicks 'Proceed Normally' then we
+  // should navigate to the originally requested page without going into
+  // off-the-record mode.
+  auto* model = browser()->tab_strip_model();
+  auto* contents = model->GetActiveWebContents();
+  auto* infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(contents);
+  TestObserver observer;
+  // Set up expectation that an infobar will NOT appear later.
+  EXPECT_CALL(observer, OnInfoBarAdded(_)).Times(0);
+  infobar_manager->AddObserver(&observer);
+
+  SetRequestOTRPref(RequestOTRService::RequestOTRActionOption::kAsk);
+  GURL url = embedded_test_server()->GetURL("sensitive.a.com", "/simple.html");
+  NavigateTo(url);
+  ASSERT_TRUE(IsShowingInterstitial());
+
+  // Simulate click on 'Proceed Normally' button. This should navigate to
+  // the originally requested page in off-the-record mode.
+  ClickAndWaitForNavigation("back-button");
+  ASSERT_FALSE(IsShowingInterstitial());
+  // Request-OTR infobar should never appear, because the user requested to
+  // proceed normally, so we should not be in off-the-record mode.
+
+  infobar_manager->RemoveObserver(&observer);
+}
+
+IN_PROC_BROWSER_TEST_F(RequestOTRBrowserTest, IncludeExclude) {
```

Comment:
I'm not sure what IncludeExclude means

Reply:
It's testing the include and exclude mechanisms within the JSON configuration file, to ensure that e.g. an excluded URL is actually excluded taking precedence over the more general inclusion rule.

Reply:
well what I'm getting at is that it needs some comments and/or a more descriptive test name :)